src/main/scala/outwatch/dom/DomTypes.scala

```diff
@@ -13,25 +13,24 @@ import cats.effect.IO
 import org.scalajs.dom
 import helpers._
 
+import scala.language.implicitConversions
+
 private[outwatch] object DomTypesBuilder {
   type GenericVNode[T] = VNode_
 
   trait VNodeBuilder extends TagBuilder[GenericVNode, VNode_] {
     // we can ignore information about void tags here, because snabbdom handles this automatically for us based on the tagname.
-    override def tag[Ref <: VNode_](tagName: String, void: Boolean): VNode_ =
-      VTree(tagName, Seq.empty)
+    override def tag[Ref <: VNode_](tagName: String, void: Boolean): VNode_ = VTree(tagName, Seq.empty)
   }
 
   object CodecBuilder {
     type Attribute[T, _] = AttributeBuilder[T]
     type Property[T, _] = PropertyBuilder[T]
 
-    def encodeAttribute[V](codec: Codec[V, String]): V => Attr.Value = codec match {
-      //The BooleanAsAttrPresenceCodec does not play well with snabbdom. it
-      //encodes true as "" and false as null, whereas snabbdom needs true/false
-      //of type boolean (not string) for toggling the presence of the attribute.
-      case _: BooleanAsAttrPresenceCodec.type => identity
-      case _ => codec.encode _
+    // snabbdom needs true/false of type boolean (not string) for toggling the presence of the attribute.
+    def encodeAttribute[V](codec: Codec[V, String]): V => Attr.Value = {
+      case b: Boolean => b
```

Comment:
This is not correct. We would interpret every boolean value as a `boolean attribute`. But only attributes with a `BooleanAsAttrPresenceCodec` codec should be rendered like this.

Forr example [contenteditable](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/contenteditable) is not a boolean attribute, but an attribute with an enumerated value (`true|false`).

Reply:
Yes, I missed that. Thanks for adding the test.